eva-server/src/main/java/uk/ac/ebi/eva/server/ws/ga4gh/beaconv2/BeaconServiceV2.java

```diff
@@ -0,0 +1,356 @@
+            });
+            variantMongo.getVariantStatsMongo().forEach(variantStatisticsMongo -> {
+                if (variantMongo.getAlternate().equalsIgnoreCase(variantStatisticsMongo.getMafAllele())) {
+                    studyIdToFrequencyMapper.put(variantStatisticsMongo.getStudyId(), variantStatisticsMongo.getMaf());
+                }
+            });
+        });
+
+        HashMap<String, VariantSource> allStudies = new HashMap<>();
+        variantSourceList.forEach(variantSource -> {
+            allStudies.put(variantSource.getStudyId(), variantSource);
+        });
+
+        allStudies.forEach((studyId, variantSource) -> {
+            if (studiesPresent.contains(studyId)) {
+                if (request.getIncludeDatasetResponses().equals(BeaconAlleleRequest.IncludeDatasetResponsesEnum.ALL) ||
+                        request.getIncludeDatasetResponses().equals(BeaconAlleleRequest
+                                .IncludeDatasetResponsesEnum.HIT)) {
+                    datasetAllelResponses.add(buildDatasetAlleleResponseHelper(true,
+                            variantSource,
+                            studyIdToFrequencyMapper.get(variantSource.getStudyId()) == null ?
+                                    null : new Float(studyIdToFrequencyMapper.get(variantSource.getStudyId()))));
+                }
+            } else {
+                if (request.getIncludeDatasetResponses().equals(BeaconAlleleRequest.IncludeDatasetResponsesEnum.ALL) ||
+                        request.getIncludeDatasetResponses().equals(BeaconAlleleRequest
+                                .IncludeDatasetResponsesEnum.MISS)) {
+                    datasetAllelResponses.add(buildDatasetAlleleResponseHelper(false,
+                            variantSource, null));
+                }
+            }
+        });
+
+        return datasetAllelResponses;
+    }
+
+    private BeaconDatasetAlleleResponse buildDatasetAlleleResponseHelper(boolean exists, VariantSource variantSource,
+                                                                         Float frequency) {
+        return new BeaconDatasetAlleleResponse()
+                .datasetId(variantSource.getStudyId())
+                .exists(exists)
+                .frequency(frequency == null ? null : new BigDecimal(frequency))
+                .variantCount(variantSource.getStats() == null ? null : (long) variantSource.getStats().
+                        getVariantsCount())
+                .sampleCount(variantSource.getStats() == null ? null : (long) variantSource.getStats().
+                        getSamplesCount())
+                .note("notestring")
+                .externalUrl("enternalUrl");
+    }
+
+    public ResponseEntity<List<BeaconAlleleResponse>> queryPost(BeaconAlleleRequest requestBody) {
+        return queryGet(requestBody.getReferenceName().toString(),
+                requestBody.getStart() == null ? null : requestBody.getStart(),
```

Comment:
This comparison doesn't change anything because it returns the value it checks so:

```suggestion
                requestBody.getStart(),
```